autoPyTorch/pipeline/components/preprocessing/tabular_preprocessing/imputation/SimpleImputer.py

```diff
@@ -15,92 +13,142 @@
 
 
 class SimpleImputer(BaseImputer):
-    """
-    Impute missing values for categorical columns with '!missing!'
-    (In case of numpy data, the constant value is set to -1, under
-    the assumption that categorical data is fit with an Ordinal Scaler)
+    """An imputer for categorical and numerical columns
+
+    Impute missing values for categorical columns with 'constant_!missing!'
+
+    Note:
+        In case of numpy data, the constant value is set to -1, under the assumption
+        that categorical data is fit with an Ordinal Scaler.
     """
 
-    def __init__(self,
-                 random_state: Optional[Union[np.random.RandomState, int]] = None,
-                 numerical_strategy: str = 'mean',
-                 categorical_strategy: str = 'most_frequent'):
+    def __init__(
+        self,
+        random_state: Optional[Union[np.random.RandomState, int]] = None,
```

Comment:
We removed integer type from `random_state`, so it must be `Optional[np.random.RandomState]`

Reply:
Okay, will do. As an fyi, this can cause different output if you run the same function twice on the same object. The random state produces a sequence of numbers.

For example, if you create a single RandomState object and pass it to every object that requires a random_state, you will get different output depending on the order in which objects use that random_state. On the flip-side, if you use an int, they are independant of each other and so it doesn't matter which order objects use it.

Reply:
Yeah, that is also true. My explanation was not sufficient, but we, in reality, decided to use `seed` for `int` and `random_state` for `np.random.RandomState`.
So it is a very good decision if we switch to `seed` instead of `random_state` in the future.

Reply:
We kept `random_state` as the arg name and allow for both, internally we just pass the `seed` argument given at construction of an `AutoSklearnClassifier` so that internally it's an int passed throughout.

We follow sklearn in principle so we copy [their expected behaviour](https://scikit-learn.org/stable/common_pitfalls.html#using-none-or-randomstate-instances-and-repeated-calls-to-fit-and-split).